hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/NoMigrationClusterStateTest.java

```diff
@@ -100,8 +100,10 @@ public void promotions_shouldHappen_whenMemberLeaves() {
         assertClusterSizeEventually(2, instances[2]);
         assertAllPartitionsAreAssigned(instances[2], 1);
 
-        assertEquals(getPartitionService(instances[1]).getPartitionStateStamp(),
-                getPartitionService(instances[2]).getPartitionStateStamp());
+        assertTrueEventually(() -> {
```

Comment:
alternatively you could `waitForClusterSafe` to ensure promotions are done before the assertion, but eventual assertion will work as well